### PR TITLE
Running End-to-end Tests on Remote Clusters

### PR DESCRIPTION
**Pull request by crimsonfaith91**
This PR creates a tutorial file about running e2e tests on remote clusters.
Partially addressing #124 

/assign @droot @pwittrock

### CODE REVIEW

File: e2e.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+**Running End-to-end Tests on Remote Clusters**
+
+This article outlines steps to run e2e tests on remote clusters for controllers created using `kubebuilder`. For example, after developing a database controller, the developer may want to run some e2e tests on a GKE cluster to verify the controller is working as expected. Currently, `kubebuilder` does not provide a template for running the e2e tests. This article serves to address this deficit.
+
+The steps are as follow:
+1.  Create a test file named `<some-file-name>_test.go` populated with template below (refer [this](https://github.com/foxish/application/blob/master/e2e/main_test.go)):
```
Comment (crimsonfaith91) [inline e2e.md]: should be `referring [this]`

File: e2e.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    // Declare a list of testing specifications and corresponding test functions
+    It("should do something", func() {
+        testDoSomething(k8sClient, roClient)
+    })
+    
+    ......
+```
+2.  Write some controller-specific e2e tests
+3.  Build controller image and upload it to an image storage website such as [gcr.io](https://cloud.google.com/container-registry/)
+4.  `go test .` to run the e2e tests
```
Comment (crimsonfaith91) [inline e2e.md]: `go test <path-to-test-file>` is better.

File: e2e.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+**Running End-to-end Tests on Remote Clusters**
+
+This article outlines steps to run e2e tests on remote clusters for controllers created using `kubebuilder`. For example, after developing a database controller, the developer may want to run some e2e tests on a GKE cluster to verify the controller is working as expected. Currently, `kubebuilder` does not provide a template for running the e2e tests. This article serves to address this deficit.
+
+The steps are as follow:
+1.  Create a test file named `<some-file-name>_test.go` populated with template below (referring [this](https://github.com/foxish/application/blob/master/e2e/main_test.go)):
```
Comment (Liujingfang1) [inline e2e.md]: Thank you for adding this file. 
Recently #229 is merged. Now there is a e2e test example inside kubebuilder you can refer
https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/e2e.go

Comment (crimsonfaith91) [reply e2e.md]: Should I just close this PR? Or should I change the PR to be based on the recently merged `e2e.go` so that we have one `e2e.md`?

**Comment by crimsonfaith91** [comment]
@droot As there is an e2e test example now, I can:
1. change `.md` file to be based on the example
2. close this PR

Which option do you think is the best?

File: e2e.md (hunk 3 above)
Comment (droot) [reply e2e.md]: This is different from the PR that got merged. PR #229 is about testing Kubebuilder end to end while this PR is about documenting steps for writing an e2e for the controller that you generated using KB. So don't close this PR.

**Comment by droot** [comment]
I am fine merging this PR as is @Liujingfang1 WDYT ?

**Comment by pwittrock** [comment]
I am good with this too.  Later we can work on scaffolding this template so users don't need to copy it.

**Comment by Liujingfang1** [comment]
> change .md file to be based on the example
> close this PR
> Which option do you think is the best?
@crimsonfaith91 Please don't close this PR, this documentation is needed.

**Comment by crimsonfaith91** [comment]
Thanks, everyone! Tell me if I can be of any other help! :)

**Pull request merged**